Tests using grunning failed

**Describe the problem**
We have disabled grunning for s390x through https://github.com/cockroachdb/cockroach/pull/95291 as patched Go runtime is not available.  Recent commit https://github.com/cockroachdb/cockroach/commit/4c2d4c39b71157799e464778db9acac931e92d49  adds tests for ARM MacOS however it also gets executed on s390x. 
On s390x, this fails with below error:
 Test:           TestRebalanceObjectiveManager
 Error Trace:    github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/rebalance_objective_test.go:209
                 Error:         Not equal:
                                expected: 1
                                actual  : 0
               
Also , recent commit (https://github.com/cockroachdb/cockroach/commit/408e227b2d23fd72869b8190a71b12674a0fdf23) to kvserver_test causing failure on s390x to TestMergeQueue/load-based-merging. 
FAIL: TestMergeQueue/load-based-merging
  --- FAIL: TestMergeQueue/load-based-merging/switch-queries-per-second-to-cpu-per-second-prev-combined-below-threshold (45.19s)
  --- FAIL: TestMergeQueue/load-based-merging/switch-cpu-per-second-to-queries-per-second-prev-combined-below-threshold (45.22s)

**To Reproduce**

Clone https://github.com/cockroachdb/cockroach
Checkout  v23.1.2
Build and execute tests using ./dev test -v

**Expected behavior**
Tests should pass.

**Environment:**
 - CockroachDB version 23.1.2
 - Server OS: Linux Ubuntu


Jira issue: CRDB-29771

Hi @Nayana-ibm, we chatted over the DB community slack.

It looks like https://github.com/cockroachdb/cockroach/pull/107710 needs to merge into release 23.1 to fix:

> On s390x, this fails with below error:
> Test: TestRebalanceObjectiveManager
> Error Trace: github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/rebalance_objective_test.go:209
> Error: Not equal:
> expected: 1
> actual : 0


In order to fix:

> Also , recent commit (408e227) to kvserver_test causing failure on s390x to TestMergeQueue/load-based-merging.
> FAIL: TestMergeQueue/load-based-merging
> --- FAIL: TestMergeQueue/load-based-merging/switch-queries-per-second-to-cpu-per-second-prev-combined-below-threshold (45.19s)
> --- FAIL: TestMergeQueue/load-based-merging/switch-cpu-per-second-to-queries-per-second-prev-combined-below-threshold (45.22s)

We need to make similar changes as #107710, to the test.

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -55,6 +55,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/grunning"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -4525,7 +4526,14 @@ func TestMergeQueue(t *testing.T) {
 
 					clearRange(t, lhsStartKey, rhsEndKey)
 					setSplitObjective(secondSplitObjective)
-					verifyUnmergedSoon(t, store, lhsStartKey, rhsStartKey)
+					if !grunning.Supported() {
+						// CPU isn't a supported split objective when grunning isn't
+						// supported. Switching the dimension will have no effect, as the
+						// objective gets overridden in such cases to always be QPS.
+						verifyMergedSoon(t, store, lhsStartKey, rhsStartKey)
+					} else {
+						verifyUnmergedSoon(t, store, lhsStartKey, rhsStartKey)
+					}
 				})
 			}
 		}

